velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -841,4 +841,106 @@ struct MillisToTimestampFunction {
   }
 };
 
+template <typename TExec>
+struct MonthsBetweenFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Timestamp>* /*date1*/,
+      const arg_type<Timestamp>* /*date2*/) {
+    timeZone_ = getTimeZoneFromConfig(config);
+  }
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Timestamp>* /*date1*/,
+      const arg_type<Timestamp>* /*date2*/,
+      const arg_type<bool>* /*roundOff*/) {
+    timeZone_ = getTimeZoneFromConfig(config);
+  }
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& /*config*/,
+      const arg_type<Timestamp>* /*date1*/,
+      const arg_type<Timestamp>* /*date2*/,
+      const arg_type<bool>* /*roundOff*/,
+      const arg_type<Varchar>* timezone) {
+    if (timezone) {
+      timeZone_ = tz::locateZone(std::string_view(*timezone), false);
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<double>& result,
+      const arg_type<Timestamp>& date1,
+      const arg_type<Timestamp>& date2) {
+    const auto datetime1 = getDateTime(date1, timeZone_);
+    const auto datetime2 = getDateTime(date2, timeZone_);
+    result = monthsBetween(datetime1, datetime2, true);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<double>& result,
+      const arg_type<Timestamp>& date1,
+      const arg_type<Timestamp>& date2,
+      const arg_type<bool>& roundOff) {
+    const auto datetime1 = getDateTime(date1, timeZone_);
+    const auto datetime2 = getDateTime(date2, timeZone_);
+    result = monthsBetween(datetime1, datetime2, roundOff);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<double>& result,
+      const arg_type<Timestamp>& date1,
+      const arg_type<Timestamp>& date2,
+      const arg_type<bool>& roundOff,
+      const arg_type<Varchar>& timezone) {
+    const auto* toTimeZone = timeZone_ != nullptr
+        ? timeZone_
+        : tz::locateZone(std::string_view(timezone), false);
+    VELOX_USER_CHECK_NOT_NULL(toTimeZone, "Unknown time zone: '{}'", timezone);
+    const auto datetime1 = getDateTime(date1, toTimeZone);
```

Comment:
Please kindly review the `getDateTime` implementation, it get the seconds and then convert to std::tm, you need both of them,  so I would suggest
```
FOLLY_ALWAYS_INLINE double monthsBetween(
      const arg_type<Timestamp>& date1,
const arg_type<Timestamp>& date2,
      timezone
      bool roundOff) {
get the seconds
get the std::tm

use the seconds to get the seconds diff
use std::tm to get the month diff
```
This will be more efficient.